lua.lib: use toLua in generateLuarocksConfig

ony opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
This is follow-up for suggestion in #225496 from @teto about re-factoring `generateLuarocksConfig` to start using newly introduced `toLua` generator.

Commit for this PR is 3e60fd2ef657ea12d03fc8d587a8d9d2e0f9c48c . Other one is for previous PR.


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


---
For `luaPackages.cqueues` generated `luarocks-config.lua` now looks like:
```
local_cache = ""
-- To prevent collisions when creating environments, we install the rock
-- files into per-package subdirectories
rocks_subdir = "cqueues-20200726-0-rocks"
-- first tree is the default target where new rocks are installed,
-- any other trees in the list are treated as additional sources of installed rocks for matching dependencies.
rocks_trees = {
  {
    ["name"] = "current",
    ["rocks_dir"] = "current",
    ["root"] = "/nix/store/2nzpqnv48b931ylxbd8y9hba5dqr6jrm-lua20200726.52-cqueues-20200726-0"
  }
}
-- For single-output external dependencies
external_deps_dirs = {
  "/nix/store/yrb14jdd2wh1xrzz4vlbsdn21vkvimsd-wrap-lua-hook",
  "/nix/store/kp355cm1wyv8dbx6rq79n823iarrq061-luarocks-3.9.1",
  "/nix/store/pi7ca6hpy4g82l96f9d8yk2k69q9wr0l-lua-5.2.4",
  "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev"
}
-- Some needed machinery to handle multiple-output external dependencies,
-- as per https://github.com/luarocks/luarocks/issues/766
variables = {
  ["CRYPTO_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
  ["CRYPTO_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
  ["CRYPTO_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib",
  ["OPENSSL_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
  ["OPENSSL_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
  ["OPENSSL_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib"
}



```
After `nix build .\#luaPackages.cqueues` can see in output of `ldd result/**/*.so`:
```
	linux-vdso.so.1 (0x00007fff591f4000)
	libssl.so.3 => /nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib/libssl.so.3 (0x00007f1b3dbc8000)
	libcrypto.so.3 => /nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib/libcrypto.so.3 (0x00007f1b3d74b000)
	libpthread.so.0 => /nix/store/8xk4yl1r3n6kbyn05qhan7nbag7npymx-glibc-2.35-224/lib/libpthread.so.0 (0x00007f1b3d746000)
	libdl.so.2 => /nix/store/8xk4yl1r3n6kbyn05qhan7nbag7npymx-glibc-2.35-224/lib/libdl.so.2 (0x00007f1b3d741000)
	librt.so.1 => /nix/store/8xk4yl1r3n6kbyn05qhan7nbag7npymx-glibc-2.35-224/lib/librt.so.1 (0x00007f1b3d73c000)
	libm.so.6 => /nix/store/8xk4yl1r3n6kbyn05qhan7nbag7npymx-glibc-2.35-224/lib/libm.so.6 (0x00007f1b3d65a000)
	libc.so.6 => /nix/store/8xk4yl1r3n6kbyn05qhan7nbag7npymx-glibc-2.35-224/lib/libc.so.6 (0x00007f1b3d451000)
	/nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib64/ld-linux-x86-64.so.2 (0x00007f1b3dccb000)
```
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ony committed:
lua.lib: use toLua in generateLuarocksConfig

ony commented:
```sh
nix shell nixpkgs\#nixpkgs-review
nixpkgs-review rev --package-regex 'lua.*Package' HEAD
```
*(limited to Lua packages alone since otherwise it planned to download ~16GiB)*

Content of `report.md`:

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>60 packages marked as broken and skipped:</summary>
  <ul>
    <li>lua51Packages.digestif</li>
    <li>lua51Packages.lmpfrlib</li>
    <li>lua51Packages.lua-resty-openidc</li>
    <li>lua51Packages.lua-resty-session</li>
    <li>lua52Packages.digestif</li>
    <li>lua52Packages.gitsigns-nvim</li>
    <li>lua52Packages.lmpfrlib</li>
    <li>lua52Packages.lua-resty-openidc</li>
    <li>lua52Packages.lua-resty-session</li>
    <li>lua52Packages.nfd</li>
    <li>lua52Packages.telescope-manix</li>
    <li>lua52Packages.telescope-nvim</li>
    <li>lua53Packages.gitsigns-nvim</li>
    <li>lua53Packages.ljsyscall</li>
    <li>lua53Packages.lua-resty-openidc</li>
    <li>lua53Packages.lua-resty-session</li>
    <li>lua53Packages.luabitop</li>
    <li>lua53Packages.nfd</li>
    <li>lua53Packages.telescope-manix</li>
    <li>lua53Packages.telescope-nvim</li>
    <li>lua54Packages.alt-getopt</li>
    <li>lua54Packages.compat53</li>
    <li>lua54Packages.cqueues</li>
    <li>lua54Packages.cyan</li>
    <li>lua54Packages.gitsigns-nvim</li>
    <li>lua54Packages.haskell-tools-nvim</li>
    <li>lua54Packages.http</li>
    <li>lua54Packages.ldoc</li>
    <li>lua54Packages.ljsyscall</li>
    <li>lua54Packages.lua-lsp</li>
    <li>lua54Packages.lua-resty-openidc</li>
    <li>lua54Packages.lua-resty-session</li>
    <li>lua54Packages.lua-zlib</li>
    <li>lua54Packages.luabitop</li>
    <li>lua54Packages.luadbi</li>
    <li>lua54Packages.luadbi-mysql</li>
    <li>lua54Packages.luadbi-postgresql</li>
    <li>lua54Packages.luadbi-sqlite3</li>
    <li>lua54Packages.luaevent</li>
    <li>lua54Packages.luaffi</li>
    <li>lua54Packages.luaposix</li>
    <li>lua54Packages.luazip</li>
    <li>lua54Packages.lush-nvim</li>
    <li>lua54Packages.luuid</li>
    <li>lua54Packages.markdown</li>
    <li>lua54Packages.nfd</li>
    <li>lua54Packages.nvim-cmp</li>
    <li>lua54Packages.plenary-nvim</li>
    <li>lua54Packages.readline</li>
    <li>lua54Packages.rest-nvim</li>
    <li>lua54Packages.sqlite</li>
    <li>lua54Packages.teal-language-server</li>
    <li>lua54Packages.telescope-manix</li>
    <li>lua54Packages.telescope-nvim</li>
    <li>lua54Packages.tl</li>
    <li>luajitPackages.digestif</li>
    <li>luajitPackages.lmpfrlib</li>
    <li>luajitPackages.lua-resty-openidc</li>
    <li>luajitPackages.lua-resty-session</li>
    <li>luajitPackages.luaffi</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>lua54Packages.bit32</li>
    <li>lua54Packages.lgi</li>
    <li>lua54Packages.lpty</li>
    <li>lua54Packages.lua-subprocess</li>
    <li>lua54Packages.vstruct</li>
  </ul>
</details>
<details>
  <summary>490 packages built:</summary>
  <ul>
    <li>lua51Packages.alt-getopt</li>
    <li>lua51Packages.argparse</li>
    <li>lua51Packages.basexx</li>
    <li>lua51Packages.binaryheap</li>
    <li>lua51Packages.bit32</li>
    <li>lua51Packages.busted</li>
    <li>lua51Packages.cassowary</li>
    <li>lua51Packages.cjson</li>
    <li>lua51Packages.cldr</li>
    <li>lua51Packages.compat53</li>
    <li>lua51Packages.cosmo</li>
    <li>lua51Packages.coxpcall</li>
    <li>lua51Packages.cqueues</li>
    <li>lua51Packages.cyan</li>
    <li>lua51Packages.cyrussasl</li>
    <li>lua51Packages.dkjson</li>
    <li>lua51Packages.fennel</li>
    <li>lua51Packages.fifo</li>
    <li>lua51Packages.fluent</li>
    <li>lua51Packages.gitsigns-nvim</li>
    <li>lua51Packages.haskell-tools-nvim</li>
    <li>lua51Packages.http</li>
    <li>lua51Packages.inspect</li>
    <li>lua51Packages.jsregexp</li>
    <li>lua51Packages.ldbus</li>
    <li>lua51Packages.ldoc</li>
    <li>lua51Packages.lgi</li>
    <li>lua51Packages.linenoise</li>
    <li>lua51Packages.ljsyscall</li>
    <li>lua51Packages.lmathx</li>
    <li>lua51Packages.loadkit</li>
    <li>lua51Packages.lpeg</li>
    <li>lua51Packages.lpeg_patterns</li>
    <li>lua51Packages.lpeglabel</li>
    <li>lua51Packages.lpty</li>
    <li>lua51Packages.lrexlib-gnu</li>
    <li>lua51Packages.lrexlib-pcre</li>
    <li>lua51Packages.lrexlib-posix</li>
    <li>lua51Packages.lua-cmsgpack</li>
    <li>lua51Packages.lua-curl</li>
    <li>lua51Packages.lua-iconv</li>
    <li>lua51Packages.lua-lsp</li>
    <li>lua51Packages.lua-messagepack</li>
    <li>lua51Packages.lua-protobuf</li>
    <li>lua51Packages.lua-resty-http</li>
    <li>lua51Packages.lua-resty-jwt</li>
    <li>lua51Packages.lua-resty-openssl</li>
    <li>lua51Packages.lua-subprocess</li>
    <li>lua51Packages.lua-term</li>
    <li>lua51Packages.lua-toml</li>
    <li>lua51Packages.lua-yajl</li>
    <li>lua51Packages.lua-zlib</li>
    <li>lua51Packages.lua_cliargs</li>
    <li>lua51Packages.luabitop</li>
    <li>lua51Packages.luacheck</li>
    <li>lua51Packages.luacov</li>
    <li>lua51Packages.luadbi</li>
    <li>lua51Packages.luadbi-mysql</li>
    <li>lua51Packages.luadbi-postgresql</li>
    <li>lua51Packages.luadbi-sqlite3</li>
    <li>lua51Packages.luaepnf</li>
    <li>lua51Packages.luaevent</li>
    <li>lua51Packages.luaexpat</li>
    <li>lua51Packages.luaffi</li>
    <li>lua51Packages.luafilesystem</li>
    <li>lua51Packages.lualdap</li>
    <li>lua51Packages.lualogging</li>
    <li>lua51Packages.luaossl</li>
    <li>lua51Packages.luaposix</li>
    <li>lua51Packages.luarepl</li>
    <li>lua51Packages.luasec</li>
    <li>lua51Packages.luasocket</li>
    <li>lua51Packages.luasql-sqlite3</li>
    <li>lua51Packages.luassert</li>
    <li>lua51Packages.luasystem</li>
    <li>lua51Packages.luaunbound</li>
    <li>lua51Packages.luaunit</li>
    <li>lua51Packages.luautf8</li>
    <li>lua51Packages.luazip</li>
    <li>lua51Packages.lush-nvim</li>
    <li>lua51Packages.luuid</li>
    <li>lua51Packages.luv</li>
    <li>lua51Packages.lyaml</li>
    <li>lua51Packages.markdown</li>
    <li>lua51Packages.mediator_lua</li>
    <li>lua51Packages.moonscript</li>
    <li>lua51Packages.mpack</li>
    <li>lua51Packages.nfd</li>
    <li>lua51Packages.nvim-client</li>
    <li>lua51Packages.nvim-cmp</li>
    <li>lua51Packages.penlight</li>
    <li>lua51Packages.plenary-nvim</li>
    <li>lua51Packages.rapidjson</li>
    <li>lua51Packages.readline</li>
    <li>lua51Packages.rest-nvim</li>
    <li>lua51Packages.say</li>
    <li>lua51Packages.serpent</li>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.std-_debug</li>
    <li>lua51Packages.std-normalize</li>
    <li>lua51Packages.stdlib</li>
    <li>lua51Packages.teal-language-server</li>
    <li>lua51Packages.telescope-manix</li>
    <li>lua51Packages.telescope-nvim</li>
    <li>lua51Packages.tl</li>
    <li>lua51Packages.vstruct</li>
    <li>lua51Packages.vusted</li>
    <li>lua52Packages.alt-getopt</li>
    <li>lua52Packages.argparse</li>
    <li>lua52Packages.basexx</li>
    <li>lua52Packages.binaryheap</li>
    <li>lua52Packages.bit32</li>
    <li>lua52Packages.busted</li>
    <li>lua52Packages.cassowary</li>
    <li>lua52Packages.cjson</li>
    <li>lua52Packages.cldr</li>
    <li>lua52Packages.compat53</li>
    <li>lua52Packages.cosmo</li>
    <li>lua52Packages.coxpcall</li>
    <li>lua52Packages.cqueues</li>
    <li>lua52Packages.cyan</li>
    <li>lua52Packages.cyrussasl</li>
    <li>lua52Packages.dkjson</li>
    <li>lua52Packages.fennel</li>
    <li>lua52Packages.fifo</li>
    <li>lua52Packages.fluent</li>
    <li>lua52Packages.haskell-tools-nvim</li>
    <li>lua52Packages.http</li>
    <li>lua52Packages.inspect</li>
    <li>lua52Packages.jsregexp</li>
    <li>lua52Packages.ldbus</li>
    <li>lua52Packages.ldoc</li>
    <li>lua52Packages.lgi</li>
    <li>lua52Packages.linenoise</li>
    <li>lua52Packages.ljsyscall</li>
    <li>lua52Packages.lmathx</li>
    <li>lua52Packages.loadkit</li>
    <li>lua52Packages.lpeg</li>
    <li>lua52Packages.lpeg_patterns</li>
    <li>lua52Packages.lpeglabel</li>
    <li>lua52Packages.lpty</li>
    <li>lua52Packages.lrexlib-gnu</li>
    <li>lua52Packages.lrexlib-pcre</li>
    <li>lua52Packages.lrexlib-posix</li>
    <li>lua52Packages.lua-cmsgpack</li>
    <li>lua52Packages.lua-curl</li>
    <li>lua52Packages.lua-iconv</li>
    <li>lua52Packages.lua-lsp</li>
    <li>lua52Packages.lua-messagepack</li>
    <li>lua52Packages.lua-protobuf</li>
    <li>lua52Packages.lua-resty-http</li>
    <li>lua52Packages.lua-resty-jwt</li>
    <li>lua52Packages.lua-resty-openssl</li>
    <li>lua52Packages.lua-subprocess</li>
    <li>lua52Packages.lua-term</li>
    <li>lua52Packages.lua-toml</li>
    <li>lua52Packages.lua-yajl</li>
    <li>lua52Packages.lua-zlib</li>
    <li>lua52Packages.lua_cliargs</li>
    <li>lua52Packages.luabitop</li>
    <li>lua52Packages.luacheck</li>
    <li>lua52Packages.luacov</li>
    <li>lua52Packages.luadbi</li>
    <li>lua52Packages.luadbi-mysql</li>
    <li>lua52Packages.luadbi-postgresql</li>
    <li>lua52Packages.luadbi-sqlite3</li>
    <li>lua52Packages.luaepnf</li>
    <li>lua52Packages.luaevent</li>
    <li>lua52Packages.luaexpat</li>
    <li>lua52Packages.luaffi</li>
    <li>lua52Packages.luafilesystem</li>
    <li>lua52Packages.lualdap</li>
    <li>lua52Packages.lualogging</li>
    <li>lua52Packages.luaossl</li>
    <li>lua52Packages.luaposix</li>
    <li>lua52Packages.luarepl</li>
    <li>lua52Packages.luasec</li>
    <li>lua52Packages.luasocket</li>
    <li>lua52Packages.luasql-sqlite3</li>
    <li>lua52Packages.luassert</li>
    <li>lua52Packages.luasystem</li>
    <li>lua52Packages.luaunbound</li>
    <li>lua52Packages.luaunit</li>
    <li>lua52Packages.luautf8</li>
    <li>lua52Packages.luazip</li>
    <li>lua52Packages.lush-nvim</li>
    <li>lua52Packages.luuid</li>
    <li>lua52Packages.luv</li>
    <li>lua52Packages.lyaml</li>
    <li>lua52Packages.markdown</li>
    <li>lua52Packages.mediator_lua</li>
    <li>lua52Packages.moonscript</li>
    <li>lua52Packages.mpack</li>
    <li>lua52Packages.nvim-client</li>
    <li>lua52Packages.nvim-cmp</li>
    <li>lua52Packages.penlight</li>
    <li>lua52Packages.plenary-nvim</li>
    <li>lua52Packages.rapidjson</li>
    <li>lua52Packages.readline</li>
    <li>lua52Packages.rest-nvim</li>
    <li>lua52Packages.say</li>
    <li>lua52Packages.serpent</li>
    <li>lua52Packages.sqlite</li>
    <li>lua52Packages.std-_debug</li>
    <li>lua52Packages.std-normalize</li>
    <li>lua52Packages.stdlib</li>
    <li>lua52Packages.teal-language-server</li>
    <li>lua52Packages.tl</li>
    <li>lua52Packages.vstruct</li>
    <li>lua52Packages.vusted</li>
    <li>lua53Packages.alt-getopt</li>
    <li>lua53Packages.argparse</li>
    <li>lua53Packages.basexx</li>
    <li>lua53Packages.binaryheap</li>
    <li>lua53Packages.bit32</li>
    <li>lua53Packages.busted</li>
    <li>lua53Packages.cassowary</li>
    <li>lua53Packages.cjson</li>
    <li>lua53Packages.cldr</li>
    <li>lua53Packages.compat53</li>
    <li>lua53Packages.cosmo</li>
    <li>lua53Packages.coxpcall</li>
    <li>lua53Packages.cqueues</li>
    <li>lua53Packages.cyan</li>
    <li>lua53Packages.cyrussasl</li>
    <li>lua53Packages.digestif</li>
    <li>lua53Packages.dkjson</li>
    <li>lua53Packages.fennel</li>
    <li>lua53Packages.fifo</li>
    <li>lua53Packages.fluent</li>
    <li>lua53Packages.haskell-tools-nvim</li>
    <li>lua53Packages.http</li>
    <li>lua53Packages.inspect</li>
    <li>lua53Packages.jsregexp</li>
    <li>lua53Packages.ldbus</li>
    <li>lua53Packages.ldoc</li>
    <li>lua53Packages.lgi</li>
    <li>lua53Packages.linenoise</li>
    <li>lua53Packages.lmathx</li>
    <li>lua53Packages.lmpfrlib</li>
    <li>lua53Packages.loadkit</li>
    <li>lua53Packages.lpeg</li>
    <li>lua53Packages.lpeg_patterns</li>
    <li>lua53Packages.lpeglabel</li>
    <li>lua53Packages.lpty</li>
    <li>lua53Packages.lrexlib-gnu</li>
    <li>lua53Packages.lrexlib-pcre</li>
    <li>lua53Packages.lrexlib-posix</li>
    <li>lua53Packages.lua-cmsgpack</li>
    <li>lua53Packages.lua-curl</li>
    <li>lua53Packages.lua-iconv</li>
    <li>lua53Packages.lua-lsp</li>
    <li>lua53Packages.lua-messagepack</li>
    <li>lua53Packages.lua-protobuf</li>
    <li>lua53Packages.lua-resty-http</li>
    <li>lua53Packages.lua-resty-jwt</li>
    <li>lua53Packages.lua-resty-openssl</li>
    <li>lua53Packages.lua-subprocess</li>
    <li>lua53Packages.lua-term</li>
    <li>lua53Packages.lua-toml</li>
    <li>lua53Packages.lua-yajl</li>
    <li>lua53Packages.lua-zlib</li>
    <li>lua53Packages.lua_cliargs</li>
    <li>lua53Packages.luacheck</li>
    <li>lua53Packages.luacov</li>
    <li>lua53Packages.luadbi</li>
    <li>lua53Packages.luadbi-mysql</li>
    <li>lua53Packages.luadbi-postgresql</li>
    <li>lua53Packages.luadbi-sqlite3</li>
    <li>lua53Packages.luaepnf</li>
    <li>lua53Packages.luaevent</li>
    <li>lua53Packages.luaexpat</li>
    <li>lua53Packages.luaffi</li>
    <li>lua53Packages.luafilesystem</li>
    <li>lua53Packages.lualdap</li>
    <li>lua53Packages.lualogging</li>
    <li>lua53Packages.luaossl</li>
    <li>lua53Packages.luaposix</li>
    <li>lua53Packages.luarepl</li>
    <li>lua53Packages.luasec</li>
    <li>lua53Packages.luasocket</li>
    <li>lua53Packages.luasql-sqlite3</li>
    <li>lua53Packages.luassert</li>
    <li>lua53Packages.luasystem</li>
    <li>lua53Packages.luaunbound</li>
    <li>lua53Packages.luaunit</li>
    <li>lua53Packages.luautf8</li>
    <li>lua53Packages.luazip</li>
    <li>lua53Packages.lush-nvim</li>
    <li>lua53Packages.luuid</li>
    <li>lua53Packages.luv</li>
    <li>lua53Packages.lyaml</li>
    <li>lua53Packages.markdown</li>
    <li>lua53Packages.mediator_lua</li>
    <li>lua53Packages.moonscript</li>
    <li>lua53Packages.mpack</li>
    <li>lua53Packages.nvim-client</li>
    <li>lua53Packages.nvim-cmp</li>
    <li>lua53Packages.penlight</li>
    <li>lua53Packages.plenary-nvim</li>
    <li>lua53Packages.rapidjson</li>
    <li>lua53Packages.readline</li>
    <li>lua53Packages.rest-nvim</li>
    <li>lua53Packages.say</li>
    <li>lua53Packages.serpent</li>
    <li>lua53Packages.sqlite</li>
    <li>lua53Packages.std-_debug</li>
    <li>lua53Packages.std-normalize</li>
    <li>lua53Packages.stdlib</li>
    <li>lua53Packages.teal-language-server</li>
    <li>lua53Packages.tl</li>
    <li>lua53Packages.vstruct</li>
    <li>lua53Packages.vusted</li>
    <li>lua54Packages.argparse</li>
    <li>lua54Packages.basexx</li>
    <li>lua54Packages.binaryheap</li>
    <li>lua54Packages.busted</li>
    <li>lua54Packages.cassowary</li>
    <li>lua54Packages.cjson</li>
    <li>lua54Packages.cldr</li>
    <li>lua54Packages.cosmo</li>
    <li>lua54Packages.coxpcall</li>
    <li>lua54Packages.cyrussasl</li>
    <li>lua54Packages.digestif</li>
    <li>lua54Packages.dkjson</li>
    <li>lua54Packages.fennel</li>
    <li>lua54Packages.fifo</li>
    <li>lua54Packages.fluent</li>
    <li>lua54Packages.inspect</li>
    <li>lua54Packages.jsregexp</li>
    <li>lua54Packages.ldbus</li>
    <li>lua54Packages.linenoise</li>
    <li>lua54Packages.lmathx</li>
    <li>lua54Packages.lmpfrlib</li>
    <li>lua54Packages.loadkit</li>
    <li>lua54Packages.lpeg</li>
    <li>lua54Packages.lpeg_patterns</li>
    <li>lua54Packages.lpeglabel</li>
    <li>lua54Packages.lrexlib-gnu</li>
    <li>lua54Packages.lrexlib-pcre</li>
    <li>lua54Packages.lrexlib-posix</li>
    <li>lua54Packages.lua-cmsgpack</li>
    <li>lua54Packages.lua-curl</li>
    <li>lua54Packages.lua-iconv</li>
    <li>lua54Packages.lua-messagepack</li>
    <li>lua54Packages.lua-protobuf</li>
    <li>lua54Packages.lua-resty-http</li>
    <li>lua54Packages.lua-resty-jwt</li>
    <li>lua54Packages.lua-resty-openssl</li>
    <li>lua54Packages.lua-term</li>
    <li>lua54Packages.lua-toml</li>
    <li>lua54Packages.lua-yajl</li>
    <li>lua54Packages.lua_cliargs</li>
    <li>lua54Packages.luacheck</li>
    <li>lua54Packages.luacov</li>
    <li>lua54Packages.luaepnf</li>
    <li>lua54Packages.luaexpat</li>
    <li>lua54Packages.luafilesystem</li>
    <li>lua54Packages.lualdap</li>
    <li>lua54Packages.lualogging</li>
    <li>lua54Packages.luaossl</li>
    <li>lua54Packages.luarepl</li>
    <li>lua54Packages.luasec</li>
    <li>lua54Packages.luasocket</li>
    <li>lua54Packages.luasql-sqlite3</li>
    <li>lua54Packages.luassert</li>
    <li>lua54Packages.luasystem</li>
    <li>lua54Packages.luaunbound</li>
    <li>lua54Packages.luaunit</li>
    <li>lua54Packages.luautf8</li>
    <li>lua54Packages.luv</li>
    <li>lua54Packages.lyaml</li>
    <li>lua54Packages.mediator_lua</li>
    <li>lua54Packages.moonscript</li>
    <li>lua54Packages.mpack</li>
    <li>lua54Packages.nvim-client</li>
    <li>lua54Packages.penlight</li>
    <li>lua54Packages.rapidjson</li>
    <li>lua54Packages.say</li>
    <li>lua54Packages.serpent</li>
    <li>lua54Packages.std-_debug</li>
    <li>lua54Packages.std-normalize</li>
    <li>lua54Packages.stdlib</li>
    <li>lua54Packages.vusted</li>
    <li>luajitPackages.alt-getopt</li>
    <li>luajitPackages.argparse</li>
    <li>luajitPackages.basexx</li>
    <li>luajitPackages.binaryheap</li>
    <li>luajitPackages.bit32</li>
    <li>luajitPackages.busted</li>
    <li>luajitPackages.cassowary</li>
    <li>luajitPackages.cjson</li>
    <li>luajitPackages.cldr</li>
    <li>luajitPackages.compat53</li>
    <li>luajitPackages.cosmo</li>
    <li>luajitPackages.coxpcall</li>
    <li>luajitPackages.cqueues</li>
    <li>luajitPackages.cyan</li>
    <li>luajitPackages.cyrussasl</li>
    <li>luajitPackages.dkjson</li>
    <li>luajitPackages.fennel</li>
    <li>luajitPackages.fifo</li>
    <li>luajitPackages.fluent</li>
    <li>luajitPackages.gitsigns-nvim</li>
    <li>luajitPackages.haskell-tools-nvim</li>
    <li>luajitPackages.http</li>
    <li>luajitPackages.inspect</li>
    <li>luajitPackages.jsregexp</li>
    <li>luajitPackages.ldbus</li>
    <li>luajitPackages.ldoc</li>
    <li>luajitPackages.lgi</li>
    <li>luajitPackages.linenoise</li>
    <li>luajitPackages.ljsyscall</li>
    <li>luajitPackages.lmathx</li>
    <li>luajitPackages.loadkit</li>
    <li>luajitPackages.lpeg</li>
    <li>luajitPackages.lpeg_patterns</li>
    <li>luajitPackages.lpeglabel</li>
    <li>luajitPackages.lpty</li>
    <li>luajitPackages.lrexlib-gnu</li>
    <li>luajitPackages.lrexlib-pcre</li>
    <li>luajitPackages.lrexlib-posix</li>
    <li>luajitPackages.lua-cmsgpack</li>
    <li>luajitPackages.lua-curl</li>
    <li>luajitPackages.lua-iconv</li>
    <li>luajitPackages.lua-lsp</li>
    <li>luajitPackages.lua-messagepack</li>
    <li>luajitPackages.lua-protobuf</li>
    <li>luajitPackages.lua-resty-http</li>
    <li>luajitPackages.lua-resty-jwt</li>
    <li>luajitPackages.lua-resty-openssl</li>
    <li>luajitPackages.lua-subprocess</li>
    <li>luajitPackages.lua-term</li>
    <li>luajitPackages.lua-toml</li>
    <li>luajitPackages.lua-yajl</li>
    <li>luajitPackages.lua-zlib</li>
    <li>luajitPackages.lua_cliargs</li>
    <li>luajitPackages.luabitop</li>
    <li>luajitPackages.luacheck</li>
    <li>luajitPackages.luacov</li>
    <li>luajitPackages.luadbi</li>
    <li>luajitPackages.luadbi-mysql</li>
    <li>luajitPackages.luadbi-postgresql</li>
    <li>luajitPackages.luadbi-sqlite3</li>
    <li>luajitPackages.luaepnf</li>
    <li>luajitPackages.luaevent</li>
    <li>luajitPackages.luaexpat</li>
    <li>luajitPackages.luafilesystem</li>
    <li>luajitPackages.lualdap</li>
    <li>luajitPackages.lualogging</li>
    <li>luajitPackages.luaossl</li>
    <li>luajitPackages.luaposix</li>
    <li>luajitPackages.luarepl</li>
    <li>luajitPackages.luasec</li>
    <li>luajitPackages.luasocket</li>
    <li>luajitPackages.luasql-sqlite3</li>
    <li>luajitPackages.luassert</li>
    <li>luajitPackages.luasystem</li>
    <li>luajitPackages.luaunbound</li>
    <li>luajitPackages.luaunit</li>
    <li>luajitPackages.luautf8</li>
    <li>luajitPackages.luazip</li>
    <li>luajitPackages.lush-nvim</li>
    <li>luajitPackages.luuid</li>
    <li>luajitPackages.luv</li>
    <li>luajitPackages.lyaml</li>
    <li>luajitPackages.markdown</li>
    <li>luajitPackages.mediator_lua</li>
    <li>luajitPackages.moonscript</li>
    <li>luajitPackages.mpack</li>
    <li>luajitPackages.nfd</li>
    <li>luajitPackages.nvim-client</li>
    <li>luajitPackages.nvim-cmp</li>
    <li>luajitPackages.penlight</li>
    <li>luajitPackages.plenary-nvim</li>
    <li>luajitPackages.rapidjson</li>
    <li>luajitPackages.readline</li>
    <li>luajitPackages.rest-nvim</li>
    <li>luajitPackages.say</li>
    <li>luajitPackages.serpent</li>
    <li>luajitPackages.sqlite</li>
    <li>luajitPackages.std-_debug</li>
    <li>luajitPackages.std-normalize</li>
    <li>luajitPackages.stdlib</li>
    <li>luajitPackages.teal-language-server</li>
    <li>luajitPackages.telescope-manix</li>
    <li>luajitPackages.telescope-nvim</li>
    <li>luajitPackages.tl</li>
    <li>luajitPackages.vstruct</li>
    <li>luajitPackages.vusted</li>
  </ul>
</details>

Manually checked out those 5 failing packages for lua 5.4 on top of current master and they fail too. See:
<details>
  <summary><code>nix build nixpkgs/master\#lua54Packages.bit32</code> (highest supported lua is 5.3)</summary>
  <pre>
       > Missing dependencies for bit32 5.3.0-1:
       >    lua >= 5.1, <= 5.3 (not installed)
       >
       > bit32 5.3.0-1 depends on lua >= 5.1, <= 5.3 (not installed)
       >
       > Error: Could not satisfy dependency lua >= 5.1, <= 5.3: Rock lua 5.4-1 is already provided by VM or via 'rocks_provided' in the config file.
  </pre>
</details>

<details>
  <summary><code>nix build nixpkgs/master\#lua54Packages.lgi</code> (sources not compatible with lua 5.4)</summary>
  <pre>
callable.c: In function 'closure_callback':
callable.c:1237:13: error: too few arguments to function 'lua_resume'
 1237 |       res = lua_resume (L, NULL, npos);
      |             ^~~~~~~~~~
In file included from lgi.h:11,
                 from callable.c:12:
/nix/store/m29gj0xgm2lf9rmqhiz4m2cblshc8h5x-lua-5.4.4/include/lua.h:300:15: note: declared here
  300 | LUA_API int  (lua_resume)     (lua_State *L, lua_State *from, int narg,
      |               ^~~~~~~~~~
make[1]: *** [Makefile:63: callable.o] Error 1
  </pre>
</details>

<details>
  <summary><code>nix build nixpkgs/master\#lua54Packages.lpty</code> (also expect lua 5.3 at most)</summary>
  <pre>
       > Missing dependencies for lpty 1.2.2-1:
       >    lua >= 5.1, <= 5.3 (not installed)
       >
       > lpty 1.2.2-1 depends on lua >= 5.1, <= 5.3 (not installed)
       >
       > Error: Could not satisfy dependency lua >= 5.1, <= 5.3: Rock lua 5.4-1 is already provided by VM or via 'rocks_provided' in the config file.
  </pre>
</details>

<details>
  <summary><code>nix build nixpkgs/master\#lua54Packages.lua-subprocess</code> (expects max lua 5.3)</summary>
  <pre>
       > Missing dependencies for subprocess scm-1:
       >    lua >= 5.1, <= 5.3 (not installed)
       >
       > subprocess scm-1 depends on lua >= 5.1, <= 5.3 (not installed)
       >
       > Error: Could not satisfy dependency lua >= 5.1, <= 5.3: Rock lua 5.4-1 is already provided by VM or via 'rocks_provided' in the config file.
  </pre>
</details>

<details>
  <summary><code>nix build nixpkgs/master\#lua54Packages.vstruct</code> (also expect lua 5.3 at most)</summary>
  <pre>
       > Missing dependencies for vstruct 2.1.1-1:
       >    lua >= 5.1, <= 5.3 (not installed)
       >
       > vstruct 2.1.1-1 depends on lua >= 5.1, <= 5.3 (not installed)
       >
       > Error: Could not satisfy dependency lua >= 5.1, <= 5.3: Rock lua 5.4-1 is already provided by VM or via 'rocks_provided' in the config file.
  </pre>
</details>

teto commented:
No worries about these broken packages, I realized that when working on https://github.com/NixOS/nixpkgs/pull/224553. 
I will merge:
1. https://github.com/NixOS/nixpkgs/pull/225496 once my comments are answered
2. https://github.com/NixOS/nixpkgs/pull/224553
3. this one (after I've run some checks with https://github.com/NixOS/nixpkgs/pull/224553)

teto commented:
If you dont mind, I'll wait for https://github.com/NixOS/nixpkgs/pull/224553 to reach unstable to see if that triggers issues (that's a PR)

teto commented on pkgs/development/lua-modules/lib.nix:
```diff
@@ -89,58 +90,50 @@ rec {
-
-      extraVariablesStr = lib.concatStringsSep "\n "
-        (lib.mapAttrsToList (k: v: "${k}='${v}';") extraVariables);
   in ''
```
why not call toLua here at the upper level ?

ony replied:
These lines are removed :slightly_smiling_face: in upper level we do:
```nix
"variables = ${toLua {} (depVariables // extraVariables)}"
```

ony replied:
Sorry. I think I misread where comment were attached to. At the moment `toLua` can only produce single Lua value and have no configuration to produce Lua content with setting multiple variables. E.g. result of 
```nix
toLua {} { variables = depVariables // extraVariables; }
```
would be something like
```lua
{
  ["variables"] = {
    ["CRYPTO_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
    ["CRYPTO_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
    ["CRYPTO_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib",
    ["OPENSSL_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
    ["OPENSSL_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
    ["OPENSSL_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib"
  }
}
```

And I'm not that familiar with Lua to know how to turn table value into bindings for local/global variables via Lua code.

ony replied:
Added configuration `asBindings` for `toLua` and lifted all calls to top expression. Note that comments no longer present in generated `luarocks-config.lua`. E.g. for `nix build '.#luaPackages.cqueues'` I get now:

```
external_deps_dirs = {
  "/nix/store/yrb14jdd2wh1xrzz4vlbsdn21vkvimsd-wrap-lua-hook",
  "/nix/store/kp355cm1wyv8dbx6rq79n823iarrq061-luarocks-3.9.1",
  "/nix/store/pi7ca6hpy4g82l96f9d8yk2k69q9wr0l-lua-5.2.4",
  "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev"
}
local_cache = ""
rocks_subdir = "cqueues-20200726-0-rocks"
rocks_trees = {
  {
    ["name"] = "current",
    ["rocks_dir"] = "current",
    ["root"] = "/nix/store/3bfvabk59yp30apbvsmnf1cq2ks9cswb-lua20200726.52-cqueues-20200726-0"
  }
}
variables = {
  ["CRYPTO_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
  ["CRYPTO_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
  ["CRYPTO_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib",
  ["OPENSSL_BINDIR"] = "/nix/store/05qna7girdhnzx38jkhr4yyh7gmnizv3-openssl-3.0.8-bin/bin",
  ["OPENSSL_INCDIR"] = "/nix/store/ggsmy3rwsr6q6ri45ck3jdqnz9cf6dm6-openssl-3.0.8-dev/include",
  ["OPENSSL_LIBDIR"] = "/nix/store/s8vg2h8xzqmjd72f3g3p1jqy2lbbapc6-openssl-3.0.8/lib"
}



```

ony committed:
lib.generators.toLua: asBindings option

Allows to generate code block for setting of global variables

ony committed:
lua.lib: lift toLua to top in generateLuarocksConfig

With asBindings option of toLua we can generate whole block of global
variables assignments.
